### ISSUE TITLE
Vite 4 support

### ISSUE THREAD

**Issue by wobo-nate** [hannoeru/unplugin-swc#34]
Any plans to upgrade the peerDependency to vite 4? This would be useful for Nest.js apps trying to use vite and vitest because esbuild doesn't support `emitDecoratorMetadata` [yet](https://github.com/evanw/esbuild/issues/104).

If anyone has a workaround to make this plugin work with vite 4, please let me know 🙏 Thanks!

**Comment by geersch** [comment]
What issue are you experiencing with Vite 4? 

I described how to use Vitest + this plugin for a NestJS app here: 

https://github.com/geersch/nestjs-getting-started/blob/main/docs/12-vitest.md

Bumped into a few issues (see caveats / known issues), but nothing related to Vite 4.

**Comment by geersch** [comment]
Hrm, I see. 

`unplugin-swc` does have a dependency on `unplugin`.

```json
{
  "dependencies": {
    "unplugin": "^0.6.0"
  },
}
```

Which resolves to version `0.6.3` as that's the latest `0.6.x` release. And that specifies version has a **peer** dependency on `vite`.

```json
{
  "peerDependencies": {
    ...
    " vite": "^2.3.0"
  }
}
```

Which indeed resolves to `2.9.16`. Using yarn 1.22.19, which did not warn me about the missing peer deps, since `vite` is marked as an optional peer dependency in `unplugin`.

https://github.com/unjs/unplugin/blob/29fe515d424101970e14c3296eb99a2d553055f7/package.json#L76

You should be able to remove that particular vite version from your `package-lock.json` / `yarn.lock` and make sure version `4.x.x` is used instead.

**Comment by hannoeru** [comment]
This should fixed in v1.4.0, give it a try!
